content/en/docs/instrumentation/go/getting-started.md

```diff
@@ -1,681 +1,1427 @@
-passing [`TracerProvider`]s may make more sense.
-
-## Putting It All Together
-
-You should now have a working application that produces trace telemetry data!
-Give it a try.
-
-```console
-$ go run .
-What Fibonacci number would you like to know:
-42
-Fibonacci(42) = 267914296
-What Fibonacci number would you like to know:
-^C
-goodbye
-```
+</details>
 
-A new file named `traces.txt` should be created in your working directory. All
-the traces created from running your application should be in there!
+## Custom instrumentation
 
-## Error handling
+Instrumentation libraries captures telemetry at the edges of your systems, such
+as inbound and outbound HTTP requests, but it doesn't capture what's going on in
+your application. For that you'll need to write some custom
+[manual instrumentation](../manual/).
 
-At this point, you have a working application and it is producing tracing
-telemetry data. Unfortunately, there is an error in the core functionality of
-the `fib` module.
+Modify `rolldice.go` to include custom instrumentation using OpenTelemetry API:
 
-```console
-$ go run .
-What Fibonacci number would you like to know:
-100
-Fibonacci(100) = 3736710778780434371
-# …
-```
+```go
+package main
 
-The 100-th Fibonacci number is `354224848179261915075`, not
-`3736710778780434371`! This application is only meant as a demo, but it
-shouldn't return incorrect values. Update the `Fibonacci` function to return an
-error instead of an incorrect value:
+import (
+	"io"
+	"log"
+	"math/rand"
+	"net/http"
+	"strconv"
 
-```go
-// Fibonacci returns the n-th Fibonacci number. An error is returned if the
-// Fibonacci number cannot be represented as a uint64.
-func Fibonacci(n uint) (uint64, error) {
-	if n <= 1 {
-		return uint64(n), nil
-	}
+	"go.opentelemetry.io/otel"
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/metric"
+)
 
-	if n > 93 {
-		return 0, fmt.Errorf("unsupported Fibonacci number %d: too large", n)
-	}
+var (
+	tracer  = otel.Tracer("rolldice")
+	meter   = otel.Meter("rolldice")
+	rollCnt metric.Int64Counter
+)
 
-	var n2, n1 uint64 = 0, 1
-	for i := uint(2); i < n; i++ {
-		n2, n1 = n1, n1+n2
+func init() {
+	var err error
+	rollCnt, err = meter.Int64Counter("roll_counter",
+		metric.WithDescription("The number of rolls by roll value"),
+		metric.WithUnit("{roll}"))
+	if err != nil {
+		panic(err)
 	}
-
-	return n2 + n1, nil
 }
-```
-
-Great, you have fixed the code, but it would be ideal to include errors returned
-to a user in the telemetry data. Luckily, spans can be configured to communicate
-such information. Update the `Write` method in `app.go` with the following code.
 
-```go
-// Write writes the n-th Fibonacci number back to the user.
-func (a *App) Write(ctx context.Context, n uint) {
-	var span trace.Span
-	ctx, span = otel.Tracer(name).Start(ctx, "Write")
+func rolldice(w http.ResponseWriter, r *http.Request) {
+	ctx, span := tracer.Start(r.Context(), "do_roll")
```

Comment:
same comment as above.  Maybe `dice.roll`?

Reply:
I will change to simply `roll` as this is the name of the operation. 

For span names I think we rather not use prefixes. See:
- https://opentelemetry.io/docs/specs/semconv/http/http-spans/#name
- https://opentelemetry.io/docs/specs/semconv/database/database-spans/

Reply:
Resolved with https://github.com/open-telemetry/opentelemetry.io/commit/f3cf8cc53f4e9d2029521be050cd0bb71c770e7b